StyleCop.Analyzers/StyleCop.Analyzers/Settings/SettingsHelper.cs

```diff
@@ -25,23 +27,67 @@ internal static class SettingsHelper
         internal const string SettingsFileName = "stylecop.json";
         internal const string AltSettingsFileName = ".stylecop.json";
 
-        private static SourceTextValueProvider<StyleCopSettings> SettingsValueProvider { get; } =
-            new SourceTextValueProvider<StyleCopSettings>(
-                text => GetSettings(options: null, tree: null, SettingsFileName, text, DeserializationFailureBehavior.ReturnDefaultSettings));
+        private static SourceTextValueProvider<Lazy<JsonValue>> JsonValueProvider { get; } =
+            new SourceTextValueProvider<Lazy<JsonValue>>(
+                text => ParseJson(text));
+
+        /// <summary>
+        /// Gets the StyleCop settings from the JSON file, i.e. without adding information frmo the .editorconfig.
+        /// </summary>
+        /// <param name="context">The context that will be used to determine the StyleCop settings.</param>
+        /// <param name="cancellationToken">The cancellation token that the operation will observe.</param>
+        /// <returns>A <see cref="SettingsFile"/> instance which contains information about the the StyleCop settings file for the given context.
+        /// Null if no settings file was found.</returns>
+        [SuppressMessage("MicrosoftCodeAnalysisPerformance", "RS1012:Start action has no registered actions", Justification = "This is not a start action")]
+        internal static SettingsFile GetStyleCopSettingsFile(this CompilationStartAnalysisContext context, CancellationToken cancellationToken)
+        {
+            return GetSettingsFile(context.Options, GetJsonValue, cancellationToken);
+
+            Lazy<JsonValue> GetJsonValue(SourceText settingsText)
```

Comment:
💡 It would be better to make this a `static` lambda in the call to `GetSettingsFile` so it can cache the delegate

Reply:
Handled in #3648